kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/ProcessInstanceQueryServicesClientImpl.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.client.impl;
+
+import static org.kie.server.api.rest.RestURI.*;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.kie.server.api.model.instance.ProcessInstance;
+import org.kie.server.api.model.instance.ProcessInstanceList;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.ProcessInstanceQueryServicesClient;
+import org.kie.server.jbpm.queries.api.model.definition.ProcessInstanceQueryFilterSpec;
+
+public class ProcessInstanceQueryServicesClientImpl extends AbstractKieServicesClientImpl implements ProcessInstanceQueryServicesClient {
+
+	public ProcessInstanceQueryServicesClientImpl(KieServicesConfiguration config) {
+		super(config);
+	}
+	
+	public ProcessInstanceQueryServicesClientImpl(KieServicesConfiguration config, ClassLoader classLoader) {
+		super(config, classLoader);
+	}
+
+	@Override
+	public List<ProcessInstance> findProcessInstancesWithFilters(ProcessInstanceQueryFilterSpec filterSpec, Integer page, Integer pageSize) {
+		ProcessInstanceList result = null;
+
+		if (config.isRest()) {
+			String queryString = getPagingQueryString("?", page, pageSize);
+			result = makeHttpPostRequestAndCreateCustomResponse(loadBalancer.getUrl() + "/" + PROCESS_INSTANCES_GET_FILTERED_URI + queryString,
```

Comment:
I need to discuss with @mswiderski, because this pattern is used all over the KIE-Server client, e.g.: https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/QueryServicesClientImpl.java

So, any changes in that pattern would need to be implemented in the whole client to keep the code consistent.